EXEC plugin panics on Freebsd  (panic: runtime error: index out of range)

I'm getting a crash with latest master build from src on freebsd 10.3 with the exec plugin:

panic: runtime error: index out of range

 ```
goroutine 28 [running]:
panic(0x10ac8c0, 0xc820010130)
    /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/telegraf/plugins/inputs/exec.(*Exec).Gather(0xc8201b6000, 0x801cb6f78, 0xc8200834a0, 0x0, 0x0)
    /root/work/src/github.com/influxdata/telegraf/plugins/inputs/exec/exec.go:181 +0x7b5
github.com/influxdata/telegraf/agent.gatherWithTimeout.func1(0xc820012660, 0xc820176660, 0xc8200834a0)
    /root/work/src/github.com/influxdata/telegraf/agent/agent.go:161 +0x73
created by github.com/influxdata/telegraf/agent.gatherWithTimeout
    /root/work/src/github.com/influxdata/telegraf/agent/agent.go:162 +0xe0
 ```

with 13.1 everything works fine. just want that zfs plugin

also tested the config path with:
 ```
    commands = ["/bin/echo"]
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,12 @@
+## v1.0 beta 2 [unreleased]
+
+### Features
+
+### Bugfixes
+
+- [#1330](https://github.com/influxdata/telegraf/issues/1330): Fix exec plugin panic when using single binary
+- [#1336](https://github.com/influxdata/telegraf/issues/1336): Fixed incorrect prometheus metrics source selection
+
 ## v1.0 beta 1 [2016-06-07]
 
 ### Release Notes
@@ -50,7 +59,6 @@ time before a new metric is included by the plugin.
 - [#1316](https://github.com/influxdata/telegraf/pull/1316): Removed leaked "database" tag on redis metrics. Thanks @PierreF!
 - [#1323](https://github.com/influxdata/telegraf/issues/1323): Processes plugin: fix potential error with /proc/net/stat directory.
 - [#1322](https://github.com/influxdata/telegraf/issues/1322): Fix rare RHEL 5.2 panic in gopsutil diskio gathering function.
-- [#1336](https://github.com/influxdata/telegraf/issues/1336): Fixed incorrect prometheus metrics source selection
 
 ## v0.13.1 [2016-05-24]
 

--- a/plugins/inputs/exec/exec.go
+++ b/plugins/inputs/exec/exec.go
@@ -177,8 +177,12 @@ func (e *Exec) Gather(acc telegraf.Accumulator) error {
 			// There were matches, so we'll append each match together with
 			// the arguments to the commands slice
 			for _, match := range matches {
-				commands = append(
-					commands, strings.Join([]string{match, cmdAndArgs[1]}, " "))
+				if len(cmdAndArgs) == 1 {
+					commands = append(commands, match)
+				} else {
+					commands = append(commands,
+						strings.Join([]string{match, cmdAndArgs[1]}, " "))
+				}
 			}
 		}
 	}

